Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/CachedKeyFileHolder.kt

```diff
@@ -97,20 +97,33 @@ object CachedKeyFileHolder {
             val deferredQueries: MutableCollection<Deferred<Any>> = mutableListOf()
             keyCache.deleteOutdatedEntries(uuidListFromServer)
             val missingDays = getMissingDaysFromDiff(serverDates)
+            val failedEntryCacheKeys = mutableListOf<String>()
             if (missingDays.isNotEmpty()) {
                 // we have a date difference
                 deferredQueries.addAll(
                     missingDays
                         .map { getURLForDay(it) }
-                        .map { url -> async { url.createDayEntryForUrl() } }
+                        .map { url ->
+                            val cacheKey = url.generateCacheKeyFromString()
+                            try {
+                                async {
+                                    url.createDayEntryForUrl(cacheKey)
+                                }
+                            } catch (e: Exception) {
```

Comment:
Hm, this doesn't work, the exception will be catched by `async`, not our try catch
We need to try catch inside the `async`.

Reply:
If we catch inside the async, the errors happen async, and `failedEntryCacheKeys` needs to be threadsafe, i.e. LinkedConcurrentQueue or a Collections.synchronized list.